### PR TITLE
Update index.rst

### PR DESCRIPTION
**Pull request by mkcor**
Should we remove this section altogether, so we don't need to update it manually anymore?

/cc @lagru @stefanv

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -52,10 +52,10 @@
 News
 ----
 
+- **Release!** Version 0.21.0 2023-06-02
 - **Release!** Version 0.20.0 2023-02-28
-- **Release!** Version 0.19.3 2022-06-12
-- `CZI announces second round of funding for scikit-image
-  <https://chanzuckerberg.com/newsroom/awards-3-8-million-open-source-software-projects-essential-science/>`__ 2020-03-27
+- `scikit-image selected for third round of funding by CZI
+  <https://chanzuckerberg.com/eoss/proposals/?cycle=5>`__ 2022-11-30
```
Comment (lagru) [inline index.rst:58]:
```suggestion
- `scikit-image selected for third round of funding by CZI
  <https://chanzuckerberg.com/eoss/proposals/from-library-to-protocol-scikit-image-as-an-api-reference/>`__ 2022-11-30
```

**Comment by lagru** [comment]
My 2cts: I like having the section even if we forget it sometimes.

File: index.rst [hunk 2]
```diff
@@ -52,10 +52,10 @@
 News
 ----
 
+- **Release!** Version 0.21.0 2023-06-02
 - **Release!** Version 0.20.0 2023-02-28
```
Comment (lagru) [inline index.rst:56]:
```suggestion
- **Release!** `Version 0.21.0 <https://scikit-image.org/docs/stable/release_notes/release_0.21.html>`_ 2023-06-02
- **Release!** `Version 0.20.0 <https://scikit-image.org/docs/0.20.x/release_notes.html>`_ 2023-02-28
```

File: index.rst (hunk 1 above)
Comment (stefanv) [reply index.rst:58]: I think two is technically correct.

File: index.rst [hunk 3]
```diff
@@ -52,10 +52,10 @@
 News
 ----
 
-- **Release!** Version 0.20.0 2023-02-28
-- **Release!** Version 0.19.3 2022-06-12
-- `CZI announces second round of funding for scikit-image
-  <https://chanzuckerberg.com/newsroom/awards-3-8-million-open-source-software-projects-essential-science/>`__ 2020-03-27
+- **Release!** `Version 0.21.0 <https://scikit-image.org/docs/stable/release_notes/release_0.21.html>`_ 2023-06-02
+- **Release!** `Version 0.20.0 <https://scikit-image.org/docs/0.20.x/release_notes.html>`_ 2023-02-28
+- `scikit-image selected for third round of funding by CZI
```
Comment (stefanv) [inline index.rst:57]: Maybe we can be a bit more explicit here: "Funding received for X as part of CZI's 5-th EOSS cycle"

File: index.rst (hunk 1 above)
Comment (mkcor) [reply index.rst:58]: > I think two is technically correct.

Yes! @lagru I didn't know either until [this conversation](https://github.com/scikit-image/scikit-image/pull/6497#discussion_r999679301) took place, where I learned from @stefanv that the double underscore is actually more correct.

File: index.rst (hunk 3 above)
Comment (mkcor) [reply index.rst:57]: Ok. What about:
"As part of CZI's 5-th EOSS grant cycle, scikit-image received funding for creating a typed, discoverable, and extensible API!"
?

Comment (stefanv) [reply index.rst:57]: That sounds great!

File: index.rst (hunk 1 above)
Comment (lagru) [reply index.rst:58]: Wow, today I learned. Good to know going forward and thanks for the link.

Comment (lagru) [reply index.rst:58]: Note, I updated the suggestion accordingly.

**Review by lagru — APPROVED** [review]
Looks good. Happy to include it this way.

File: index.rst [hunk 4]
```diff
@@ -52,10 +52,11 @@
...
 ----
 
-- **Release!** Version 0.20.0 2023-02-28
-- **Release!** Version 0.19.3 2022-06-12
-- `CZI announces second round of funding for scikit-image
-  <https://chanzuckerberg.com/newsroom/awards-3-8-million-open-source-software-projects-essential-science/>`__ 2020-03-27
+- **Release!** `Version 0.21.0 <https://scikit-image.org/docs/stable/release_notes/release_0.21.html>`__ 2023-06-02
+- **Release!** `Version 0.20.0 <https://scikit-image.org/docs/0.20.x/release_notes.html>`__ 2023-02-28
+- As part of CZI's 5th EOSS grant cycle, scikit-image received funding for
+  `creating a typed, discoverable, and extensible API
```
Comment (lagru) [inline index.rst:58]: Actually, to me this sounds better... :smile: 
```suggestion
- As part of CZI's 5th EOSS grant cycle, scikit-image received funding to
  `create a typed, discoverable, and extensible API
```

**Pull request merged**